### PR TITLE
noImplicitAny: 1670 errors

### PR DESCRIPTION
**Pull request by tskarhed**
#14714

Roughly 700 fixes. Idk, might do more later. Down to [1670](https://en.wikipedia.org/wiki/1670) errors.

### CODE REVIEW

File: public/app/features/templating/template_srv.ts [hunk 1]
```diff
@@ -266,19 +266,19 @@ export class TemplateSrv {
     });
   }
 
-  isAllValue(value) {
+  isAllValue(value: any) {
     return value === '$__all' || (Array.isArray(value) && value[0] === '$__all');
   }
 
-  replaceWithText(target, scopedVars) {
+  replaceWithText(target: any, scopedVars: ScopedVars) {
```
Comment (torkelo) [inline public/app/features/templating/template_srv.ts]: target should be a string

File: public/app/features/users/UsersListPage.tsx [hunk 2]
```diff
@@ -59,13 +59,13 @@ export class UsersListPage extends PureComponent<Props, State> {
...
 
-  onRoleChange = (role, user) => {
+  onRoleChange = (role: any, user: any) => {
     const updatedUser = { ...user, role: role };
 
     this.props.updateUser(updatedUser);
   };
 
-  onRemoveUser = user => {
+  onRemoveUser = (user: any) => {
```
Comment (torkelo) [inline public/app/features/users/UsersListPage.tsx]: should be OrgUser I think?

File: public/app/features/users/UsersListPage.tsx [hunk 3]
```diff
@@ -59,13 +59,13 @@ export class UsersListPage extends PureComponent<Props, State> {
     return await this.props.loadInvitees();
   }
 
-  onRoleChange = (role, user) => {
+  onRoleChange = (role: any, user: any) => {
```
Comment (torkelo) [inline public/app/features/users/UsersListPage.tsx]: user should be OrgUser, and think we have a enum or type for OrgRole

File: public/app/features/users/state/selectors.ts [hunk 4]
```diff
@@ -1,18 +1,18 @@
-export const getUsers = state => {
+export const getUsers = (state: any) => {
```
Comment (torkelo) [inline public/app/features/users/state/selectors.ts]: State should be StoreState I think

File: public/app/features/users/state/selectors.ts [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
   const regex = new RegExp(state.searchQuery, 'i');
 
-  return state.users.filter(user => {
+  return state.users.filter((user: any) => {
```
Comment (torkelo) [inline public/app/features/users/state/selectors.ts]: this looks strange, if users array is typed there should be no need for any here

File: public/app/features/users/state/selectors.ts [hunk 6]
```diff
@@ -1,18 +1,18 @@
...
     return regex.test(user.login) || regex.test(user.email);
   });
 };
 
-export const getInvitees = state => {
+export const getInvitees = (state: any) => {
```
Comment (torkelo) [inline public/app/features/users/state/selectors.ts]: StoreState

File: public/app/features/users/state/selectors.ts [hunk 7]
```diff
@@ -1,18 +1,18 @@
...
   const regex = new RegExp(state.searchQuery, 'i');
 
-  return state.invitees.filter(invitee => {
+  return state.invitees.filter((invitee: any) => {
```
Comment (torkelo) [inline public/app/features/users/state/selectors.ts]: is filter array function not a natively typed (generically)  in typescript ?

File: public/app/features/users/state/selectors.ts [hunk 8]
```diff
@@ -1,18 +1,18 @@
...
     return regex.test(invitee.name) || regex.test(invitee.email);
   });
 };
 
-export const getInviteesCount = state => state.invitees.length;
-export const getUsersSearchQuery = state => state.searchQuery;
+export const getInviteesCount = (state: any) => state.invitees.length;
+export const getUsersSearchQuery = (state: any) => state.searchQuery;
```
Comment (torkelo) [inline public/app/features/users/state/selectors.ts]: StoreState or UserState?

File: public/app/plugins/datasource/cloudwatch/config_ctrl.ts [hunk 9]
```diff
@@ -71,14 +72,14 @@ export class CloudWatchConfigCtrl {
   getRegions() {
     this.datasourceSrv
       .loadDatasource(this.current.name)
-      .then(ds => {
+      .then((ds: any) => {
```
Comment (torkelo) [inline public/app/plugins/datasource/cloudwatch/config_ctrl.ts]: CloudWatchDatasource?

File: public/app/plugins/datasource/cloudwatch/datasource.ts [hunk 10]
```diff
@@ -205,21 +205,21 @@ export default class CloudWatchDatasource extends DataSourceApi<CloudWatchQuery>
     return this.doMetricQueryRequest('namespaces', null);
   }
 
-  getMetrics(namespace, region) {
+  getMetrics(namespace: string, region: any) {
```
Comment (torkelo) [inline public/app/plugins/datasource/cloudwatch/datasource.ts]: region is string as well?

File: public/app/plugins/datasource/cloudwatch/datasource.ts [hunk 11]
```diff
@@ -205,21 +205,21 @@ export default class CloudWatchDatasource extends DataSourceApi<CloudWatchQuery>
...
     return this.doMetricQueryRequest('metrics', {
       region: this.templateSrv.replace(this.getActualRegion(region)),
       namespace: this.templateSrv.replace(namespace),
     });
   }
 
-  getDimensionKeys(namespace, region) {
+  getDimensionKeys(namespace: string, region: any) {
```
Comment (torkelo) [inline public/app/plugins/datasource/cloudwatch/datasource.ts]: region string (here and every where in this file

**Review by torkelo — COMMENTED** [review]
Great progress , found some minor improvements

File: public/app/features/users/state/selectors.ts (hunk 7 above)
Comment (dehrax) [reply public/app/features/users/state/selectors.ts]: Yeah, it is, I think I just went a little bit on autopilot here

**Pull request merged**